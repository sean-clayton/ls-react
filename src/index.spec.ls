require! {
  './index.ls': {$, h, h1, h2, p}
  '@emotion/styled': {default: styled}
  'styled-components': {default: styledc}
  'react-test-renderer': render
  'react': {use-state, use-effect}
  'react-dom': ReactDOM
  'react-dom/test-utils': {act}
}

describe 'h' !->
  test 'can render elements' !->
    component = render.create do
      h 'p' {} 'Yoo!'
    
    tree = component.toJSON()

    expect tree .to-match-snapshot()

  test 'can render elements with props' !->
    component = render.create do
      h 'p' class-name: 'thing', 'Yoo!'

    tree = component.toJSON()

    expect tree .to-match-snapshot()

describe '$' !->
  test 'it works' !->
    component = render.create do
      $ do
        h1 {} 'Heading 1'
        h2 {} 'Heading 2'
        p {} 'Paragraph'

    tree = component.toJSON()

    expect tree .to-match-snapshot()

describe 'hooks' !->
  test 'they work' !->
    App = ->
      [ctr, setCtr] = use-state 0
      use-effect do
        !->
          setCtr 1
        []
      ctr

    el = document.create-element 'div'

    act !->
      ReactDOM.render do
        h App, {} null
        el

    expect el.innerHTML .toBe '1'

describe 'css-in-ls' !->
  test 'emotion' !->
    StyledButton = styled("button") do
      background-color: \red

    component = render.create do
      h StyledButton, {}, 'Click me!'

    tree = component.toJSON()
    
    expect tree .to-match-snapshot()

  test 'styled-components' !->
    StyledButton = styledc.button do
      background-color: \papayawhip

    component = render.create do
      h StyledButton, {}, 'Click me!'

    tree = component.toJSON()

    expect tree .to-match-snapshot()
