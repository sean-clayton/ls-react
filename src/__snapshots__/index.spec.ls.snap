// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`$ it works 1`] = `
Array [
  <h1>
    Heading 1
  </h1>,
  <h2>
    Heading 2
  </h2>,
  <p>
    Paragraph
  </p>,
]
`;

exports[`css-in-ls emotion 1`] = `
<button
  className="css-f5yfbg"
>
  Click me!
</button>
`;

exports[`css-in-ls styled-components 1`] = `
<button
  className="sc-bdVaJa YQOsB"
>
  Click me!
</button>
`;

exports[`h can render elements 1`] = `
<p>
  Yoo!
</p>
`;

exports[`h can render elements with props 1`] = `
<p
  className="thing"
>
  Yoo!
</p>
`;
